src/main/java/com/google/devtools/build/lib/bazel/bzlmod/SelectionFunction.java

```diff
@@ -0,0 +1,81 @@
+package com.google.devtools.build.lib.bazel.bzlmod;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
+import com.google.devtools.build.skyframe.SkyFunction;
+import com.google.devtools.build.skyframe.SkyFunctionException;
+import com.google.devtools.build.skyframe.SkyKey;
+import com.google.devtools.build.skyframe.SkyValue;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+public class SelectionFunction implements SkyFunction {
+
+  @Override
+  public SkyValue compute(SkyKey skyKey, Environment env)
```

Comment:
Do I understand correctly that this takes `DiscoveryValue` and runs an algorithm on it that doesn't depend on anything else? In that case, consider folding this into `DiscoveryFunction` because there is no reason to have two SkyFunctions. The reasons are usually:

* Caching (doesn't apply because it's not possible for `DiscoveryValue` to be cached but `SelectionValue` to be invalid)
* Breaking up computation so that we don't hog a CPU (doesn't look like it applies because the computation isn't that long)